### ISSUE TITLE
USBMS screen battery level

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#10097]
This is more of a "Quality of life" improvement and I am sure I am not the only person affected by this.

**Context:**
- I charge my Kobo at night and update my reading list over the USB while it is charging
- I limit my battery charging to 80 percent to preserve its lifetime

**Problem:**
USBMS screen does a number of things which create the problem of this issue:
1. Turns off the screen light
2. Inverts the screen
3. Disables the "battery charge limit warning" dialog

So currently I have to check a small battery percentage continuously (3) on an inverted screen (2) without a backlight (1).
To do this I have to take the reader to angle it to the computer screen to be able to see the battery percentage. But it doesn't have to be like this because we have the technology !

**Solution:**
Maybe that giant USB logo can be replaced with an equally sized battery percentage as many phones do for the exact same reason: **Seeing your charge level with a glance even from across the room**

**Exhibit A:**

![phone charging screen](https://user-images.githubusercontent.com/84420582/217682040-751abfa2-9417-4c35-a4b9-5e419798bea2.jpg)

**Comment by NiLuJe** [comment]
Unlikely to change, this isn't a phone; micromanaging charging is unhelpful (and the actual gauge is basically a software lie anyway). The charge level in the USBMS screen is basically there as a mildly helpful "nice to have" information, that's it.

----

Sidebar:

> * I limit my battery charging to 80 percent to preserve its lifetime

That is entirely unhelpful, FWIW ;). (see above re: the software gauge is lying to you).

**Comment by offset-torque** [comment]
> micromanaging charging is unhelpful at best

That sentence should start with "I believe...". Because the longevity effect of limiting li-ion chemistry battery charge-discharge levels is such an established fact that all Electric Vehicle manufacturers integrate this directly into their Battery Management Systems. Samsung phones has this, Dell and Lenovo laptops has charge limit in the BIOS. Also I have read the studies about this which points to exactly same conclusions. Also I have a 5 year old phone which still keeps 87% of its original battery capacity. 
So in short "Yes, limiting the charge level works".

> the software gauge is lying to you

Probably. But it's still useful as a heuristic. I assume you know how li-ion batteries charge and I am trying to avoid the "constant voltage" region. And 80 percent is definitely under that so it is a good indicator for me to unplug and gives me plenty of reading time.

> Unlikely to change

So you can't make it even a bit bigger ? Because I see plenty of margin in/over that line.
Or maybe you can stop inverting the screen ?
Anything to help me see my useless charging level better ?

**Comment by NiLuJe** [comment]
What 100% gauge reflects is a moving target that is *already* handled for you behind the scenes. You don't really gain much by micro-managing further, unless you intend to ship or store the device.

> So you can't make it even a bit bigger

On which device would that be? It's pretty close over here (to the point that I *can* get it to spill over two lines on a Sage, actually. Admittedly, only with the PowerCover, which nearly makes it a non-issue as the PowerCoiver breaks USBMS in fun and interesting ways).

> Or maybe you can stop inverting the screen ?

Definitely not in the cards either ;).

**Comment by NiLuJe** [comment]
I played a bit with font sizes, and I'm not particularly enthused... ;).

Current:

![current](https://user-images.githubusercontent.com/111974/217734413-58c6c5cd-d3e5-4049-9023-312f112281b1.png)

Best-fit, with the message font matched:

![fit](https://user-images.githubusercontent.com/111974/217734489-5ab80788-2e5d-4c2a-ada6-25d8cf0d24c9.png)

Best-fit, unmatched:

![fit_r2](https://user-images.githubusercontent.com/111974/217734529-3257389d-8efc-4560-b396-f23c6ec92492.png)

----

TL;DR:

* Despite running out of *vertical* space, it already looks way too big for my taste :/
* Matching the size with the message font is problematic, as we *also* run out of vertical space for the message itself (as is evident in the screenshot, which isn't even the most problematic example of this (the fuser report on a busy mount would probably be worse).
* Not matching the size looks weird as all hell.

**Comment by NiLuJe** [comment]
> > Or maybe you can stop inverting the screen ?
> 
> Definitely not in the cards either ;).

But, on devices with hardware nightmode (i.e., everything except the Sage & Elipsa) you can fudge it yourself by bracketing the USBMS session in a global nightmode toggle (that will essentially invert the... inversion ;p.)

e.g., run `fbdepth -H toggle` before and after an USBMS session.

**Comment by offset-torque** [comment]
> Definitely not in the cards either ;)

I never ever use night mode on my device, why not respect my setting. Not cool (for a software).

> Not matching the size looks weird as all hell

Looks nice to me. More important items get bigger or bolder (since we can't do color). Basic UI principle.

You are holding your (strictly personal) aesthetics over usability issues **NiLuJe** but I won't insist more since this is not that frequent or big problem. Thanks for spending effort on screenshots at least.

**Comment by NiLuJe** [comment]
> More important items

But they're not ;). If anything, the message area is what I consider key.

**Comment by NiLuJe** [comment]
> I never ever use night mode

Neither do I; it just happens to be an easy technical shortcut to achieve the wanted effect of dimming the screen, because it's no longer important, what is is the device it's plugged to.

The only concession I can see making here is not dimming it when plugged to a dedicated charger, as it would still honor that reasoning.

**Comment by NiLuJe** [comment]
> not dimming the screen when plugged to a dedicated charger

Which, err, might already be the case now that I think about it ;p (can't check, on mobile).

**Comment by offset-torque** [comment]
> Neither do I; it just happens to be an easy technical shortcut to achieve the wanted effect of dimming the screen, because it's no longer important, what is is the device it's plugged to.

Ok, kind of makes sense. At least there is a reasoning instead of you just liking the black color :)

> The only concession I can see making here would be not dimming the screen when plugged to a dedicated charger, as it would still honor that reasoning.

I didn't offer this because this will probably annoy more people than it helps. Since you can't turn off the light on USBMS screen, people will have to unplug to turn it off. So probably not a good idea.
And yes it doesn't change the light on dedicated charger (which is the expected behavior). I just checked. 

You don't need to spend any effort on this **NiLuJe**. As I said it's not a daily problem, just sometimes it is annoying. Thanks for the explanations.
